modules/swagger-codegen/src/main/resources/scala/api.mustache

```diff
@@ -53,27 +53,57 @@ class {{classname}}(val defBasePath: String = "{{basePath}}",
     {{/isPrimitiveType}}
     {{/required}}
     {{/allParams}}
-    {{#queryParams}}if(String.valueOf({{paramName}}) != "null") queryParams += "{{baseName}}" -> {{paramName}}.toString
+    {{#queryParams}}
+    {{#required}}
+    if(String.valueOf({{paramName}}) != "null") queryParams += "{{baseName}}" -> {{paramName}}.toString
```

Comment:
I don't think that's the correct way to do a null check in scala

It would be nice to to do the following:
- add an else clause to throw an exception (e.g. java.lang.IllegalArgumentException)
- add the same null check on required parameter for other type of parameters (e.g. header, form, body)